pkg/config/tlscfg/flags_test.go

```diff
@@ -173,35 +187,51 @@ func TestFailedTLSFlags(t *testing.T) {
 		t.Run(metaTest.side, func(t *testing.T) {
 			for _, test := range metaTest.tests {
 				t.Run(test, func(t *testing.T) {
-					type UnderTest interface {
-						AddFlags(flags *flag.FlagSet)
-						InitFromViper(v *viper.Viper) (Options, error)
-					}
-					var underTest UnderTest
+					var (
+						addFlags      func(*flag.FlagSet)
+						initFromViper func(*viper.Viper) (interface{}, error)
+					)
+
 					if metaTest.side == "client" {
-						underTest = &ClientFlagsConfig{
-							Prefix: "prefix",
+						clientConfig := &ClientFlagsConfig{Prefix: "prefix"}
+						addFlags = clientConfig.AddFlags
+						initFromViper = func(v *viper.Viper) (interface{}, error) {
+							return clientConfig.InitFromViper(v)
 						}
 					} else {
-						underTest = &ServerFlagsConfig{
-							Prefix: "prefix",
+						serverConfig := &ServerFlagsConfig{Prefix: "prefix"}
+						addFlags = serverConfig.AddFlags
+						initFromViper = func(v *viper.Viper) (interface{}, error) {
+							return serverConfig.InitFromViper(v)
 						}
 					}
-					v, command := config.Viperize(underTest.AddFlags)
+
+					v, command := config.Viperize(addFlags)
 
 					cmdLine := []string{
 						"--prefix.tls.enabled=true",
 						"--prefix.tls" + test,
 					}
 					err := command.ParseFlags(cmdLine)
 					require.NoError(t, err)
-					_, err = underTest.InitFromViper(v)
+
+					result, err := initFromViper(v)
 					require.NoError(t, err)
 
+					switch metaTest.side {
+					case "client":
+						_, ok := result.(configtls.ClientConfig)
+						require.True(t, ok, "expected result to be of type configtls.ClientConfig")
```

Comment:
there are functions in testify that directly test for type, which makes for a better error message than "expected to be true"